homework/src/index.js

```diff
@@ -22,25 +23,114 @@
     Object.keys(options).forEach((key) => {
       const value = options[key];
       if (key === 'text') {
-        elem.innerText = value;
+        elem.innerHTML = value;
       } else {
         elem.setAttribute(key, value);
       }
     });
     return elem;
   }
 
+  const root = document.getElementById('root');
+  const div = createAndAppend('div', root, { class: 'data' });
+  const header = createAndAppend('p', div, { text: 'HYF Repositories', class: 'header' });
+  const select = createAndAppend('select', header, { id: 'list' });
+  const table = createAndAppend('table', root, { class: 'table' });
+  const tbody = createAndAppend('tbody', table);
+  const tr = createAndAppend('tr', tbody, { text: 'Repository: ', class: 'label' });
+
+  const repositoryLink = createAndAppend('a', tr);
+  const description = createAndAppend('tr', tbody, { class: 'label' });
+  const forks = createAndAppend('tr', tbody, { class: 'label' });
+  const container = createAndAppend('div', root, { id: 'container' });
+  const updated = createAndAppend('tr', tbody, { class: 'label' });
```

Comment:
Lines 34-46 are executed immediately when `index.js` is loaded. It is best to postpone execution until the web page is fully loaded. This happens on the `load` event handled by the `windows.onload` event handler. In our case this is function `main()`. It is therefore best to move these lines inside the function `main()`, just before the first `fetchJSON()`. This could also mean that you may need to pass some of these variables as parameters to your other functions, as they are. no longer globally accessible. Using parameters instead of global variables makes your code more maintainable as you can directly what data the function needs by looking at its parameter list.